### PR TITLE
Fix all assertion failures using debug SpiderMonkey

### LINKED ISSUE #41: Add debugmozjs feature
This will make it easier to test josephine with a debug build of spidermonkey, which contains many useful assertions.

### PR DESCRIPTION
**Pull request by jdm**
This fixes #41 and #42.

### CODE REVIEW

File: examples/minidom/fake_codegen.rs [hunk 1]
```diff
@@ -71,15 +74,15 @@ static WINDOW_CLASS: JSClass = JSClass {
         mayResolve: None,
         resolve: None,
         setProperty: None,
-        trace: Some(trace_jsobject_with_native_data),
+        trace: Some(JS_GlobalObjectTraceHook),
```
Comment (asajeffrey) [inline examples/minidom/fake_codegen.rs:77]: OK, we're into jsapi weirdness here, why does this pick up josephine's object tracer? Does this assume that the global was created by josephine?

**Review by asajeffrey — COMMENTED** [review]
All looks very sensible, just one question.

File: examples/minidom/fake_codegen.rs (hunk 1 above)
Comment (jdm) [reply examples/minidom/fake_codegen.rs:77]: The JSAPI requires the global object to use JS_GlobalObjectTraceHook, which does some important stuff and then delegates to the trace hook specified in the CompartmentOptions that were provided when the global was created. The new method I've added to JSInitializer to provide a global trace hook defaults to trace_jsobject_with_native_data, so it should still be invoked for the global object.

Comment (asajeffrey) [reply examples/minidom/fake_codegen.rs:77]: Okay, that makes sense.

**Comment by asajeffrey** [comment]
If you resolve the conflicts, I'll squash and merge.

**Comment by jdm** [comment]
Done.

**Pull request merged**